src/services/categoryService.js

```diff
@@ -0,0 +1,22 @@
+const categoryDao = require('../models/categoryDao');
+const error = require('../utils/error');
+
+const getCategory = async (type) => {
+  let categoryIds = [1, 2, 4];
+  let result = []
+  if (type === '지출') {
+    for (let i = 0; i < 3; i++) {
+      let category = await categoryDao.getCategory(categoryIds[i]);
+      result[i] = await category[0];
+      result[i].type = type;
+    }
+  }
+  else if (type === '수입') {
+    let category = await categoryDao.getCategory(3)
+    result[0] = await category[0]
+    result[0].type = type;
+  }
+  return result;
+}
+
+module.exports = { getCategory}
```

Comment:
```suggestion
const getCategory = async (type) => {
  const categoryTypes = {
    '지출' : [1, 2, 4],
    '수입' : [3]
  }
  
  const categories = await categoryDao.getCategoriesByIds(categoryTypes[type]);
  categories.map((category) => { category.type = type })
  
  return categories
}

module.exports = { getCategory }
```

1. 수입/지출의 타입과 categoryId를 모두 하드코딩 해두기 보다는 타입으로 모아서 정리하시는 방법이 보다 확장성 있고, 가독성도 높을 것 같습니다.

2. category를 dao에서 하나씩 for loop으로 getCategory dao에 query를 날리시기보다는 getCategoriesByIds 이라는 dao 함수를 만드셔서 배열을 통째로 인자로 보내는 것이 좋을 것 같습니다.
그래서 dao단 SQL 쿼리문에서도 인자를 배열로 받으셔서 여러 카테고리를 한 번에 SELECT 하는 방식을 추천드립니다.
Query 문을 보내는 횟수를 3회에서 1회로 줄이기 위함입니다

Reply:
3. 근본적인 질문이 하나 있는데, category가 flow_type_id를 FK로 가지고 있으면, 하드코딩을 안해도 될 것 같은데 FK 생성하시는게 좋지 않을까요? 이부분은 백엔드 팀 같이 라운지에서 저 한 번 찾아와 주세요!